### DISCUSSION TITLE
What's the Development Plan for Rust Crates?

### DISCUSSION THREAD

**Discussion by sajvn0x** [ceylonai/ceylon#17] [Q&A] [answered]
The codebase has developed two Rust crates: `libs/sangedama` and `bindings/ceylon`.

As indicated by the folder name, the _libs/sangedama_ crate was intended to serve as a library source. However, it contains a *main.rs* file, which Cargo interprets as a binary crate. This raises questions about the intended purpose and future plans for these crates.

Could you provide a detailed explanation of the objectives and roadmap for these Rust crates? I am glad to help with these crates.

**Comment by dewmal** [comment, marked as answer]
`bindings/ceylon` is the base project for Ceylon. Currently, it contains only Python bindings. However, in the future, we plan to release it for Kotlin and Swift.

`libs/sangedama` is the communication layer for the agents. We are planning to remove main.rs from it. We have to separate these two to make it easier to manage language bindings in the future
